core/benches/kura.rs

```diff
@@ -42,12 +45,10 @@ async fn measure_block_size_for_n_validators(n_validators: u32) {
         iroha_core::kura::Kura::new(iroha_config::kura::Mode::Strict, dir.path(), false).unwrap();
     let _thread_handle = iroha_core::kura::Kura::start(kura.clone());
 
-    let block = PendingBlock::new(vec![tx], Vec::new())
-        .chain_first()
-        .validate(
-            &TransactionValidator::new(transaction_limits),
-            &Arc::new(WorldStateView::new(World::new(), kura)),
-        );
+    let block = BlockBuilder::new(vec![tx], Topology::new(Vec::new()), Vec::new()).chain_first(
```

Comment:
How does this improve the API?

Reply:
I now know less about the meanings of the other objects. Are you sure that positional arguments here make sense?

Reply:
> I now know less about the meanings of the other objects.

What information that was obvious to you in the previous version of the code are you now missing? I don't see much difference with respect to the readability

I suppose I had a different perspective in mind. With builders I'm differentiating between required and optional arguments. To create a block you always have to provide list of transactions, a topology and a list of event recommendations, then you chain the block (either as genesis or with the previous block) and then you sign it. At this point I say you have constructed a block.

>  Are you sure that positional arguments here make sense?

can I get a recommendation of what you would do here?

Reply:
> I don't see much difference with respect to the readability

Why change something if it doesn't improve readability. Especially if something close to it could. 

> What information that was obvious to you in the previous version of the code are you now missing?

`PendingBlock` is constructed. 

In the new version, `Topology` is explicitly constructed from a sensible default, which raises the question, why wasn't it implicit (as before). 

> With builders ...

Which would look more like 

```suggestion
    PendingBlockBuilder::new(vec![tx])
      .topology(Vec::new().into())
      .events(Vec::new())
      .build()?
      .chain_first(
```

> ... I'm differentiating between required and optional arguments (that have some sensible default)

By that definition only one argument is required: the topology in this example is constructed from an empty vector (which can be considered its `Default`) and event recommendations are empty, which is also a natural and sensible default. Given you already have builders and a run-time check to convert to a block, the following would be a viable API: 

```rust
    PendingBlockBuilder{
      txn: vec![tx], 
      ..Default::default()
      // topology: Topology::new(Vec::new()), // implied
      // events: Vec::new(),  // implied
    }
    .check()? 
    .chain_first(
```
Which could make things a bit more readable. As is, I don't see why it couldn't have also been 

```suggestion
    pending_block(vec![tx], vec![].into(), Vec::new()).chain_first(
```

or better yet: 

```suggestion
    PendingBlock::from(tx).chain_first(
```

## Disclaimer

This is a very minor point. 

I'd rather we merged the code with this minor annoyance to be fixed later (or never) than spend another day with 17+ unmerged PRs.